### DISCUSSION TITLE
Passing --help to custom subcommand

### DISCUSSION THREAD

**Discussion by vermiculus** [git-for-windows/git#3553] [Q&A] [answered]
I've written a custom subcommand, `git-foobar`, and have placed it in `$PATH` such that it can be called as `git-foobar`.  However, when I call `git foobar --help`, I'm greeted with the following error:

> Windows cannot find 'C:/.../Git/mingw64/share/doc/git-doc/git-foobar.html'. Make sure you typed the name correctly, and then try again.

If this file doesn't exist, why isn't `--help` simply being passed to `git-foobar`? Is there a way I can make this happen with some kind of configuration / renaming? From my read of [the implementation](https://github.com/git-for-windows/git/blob/main/builtin/help.c#L585), it doesn't look like it -- but I thought I'd ask before pursuing a patch.)

**Comment by rimrul** [comment, marked as answer]
`git foo --help` gets internally converted to `git help foo` The `git help` command works by deciding a help page to show and then passing that to `info`/`man`(or some other man page viewer)/a browser (or a helper application) to show that page, depending on configuration. Local HTML help pages are the only case where we can even detect that the page doesn't exist, and even that detection is new in Git 2.34.0 (or maybe 2.33.1).

The message you've got is different from the message Git would produce, so I'm assuming you're on an older version.

You could propose passing the `--help` option to external commands in this specific case on the mailing list, but I'm not sure they'll like that idea. This only works in a certain edge case (that's admittedly the default situation on Git for Windows and thus a rather common edge case). 

It would be nice to have a more general solution for external commands, but I'm not sure how that would look.

**Reply by dscho** [reply]
Indeed, it would be nice for a more general solution. As this is not Windows-specific, I would suggest to take it to [the Git mailing list](mailto:git@vger.kernel.org) (send plain-text messages, HTML messages are dropped silently).

**Reply by vermiculus** [reply]
I've started a discussion upstream: https://lore.kernel.org/git/CABceR4ZW4rRWZnH0ZBkWty_H84Z4CmXque_LO+1edETEWrO8PQ@mail.gmail.com/T/#u

Thanks folks!

**Comment by CervEdin** [comment]
Interesting!

A temporary work-around is to use `-h` only, or call as `git-foo --help`.

On a tangential note, you can't `git grep '--help'` 🥳

**Comment by CervEdin** [comment]
In the context of this project, it means that `git update-git-for-windows --help` doesn't work

**Reply by dscho** [reply]
That's not necessarily the same issue. As the `git-update-git-for-windows` script _is_ provided by the Git for Windows installer, there is no reason why the respective `.html` page could not be included, either. The problem _there_ is that nobody volunteered to open a PR that provides said `.html` page.
